Client

tudor-dragan opened the pull request:
The final version of the project.

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

kxv26 committed:
tooltips and file acceptance

kxv26 committed:
tooltips and file acceptance

kxv26 committed:
switching pcs, ignore

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
test

bys1 committed:
add icon

bys1 committed:
add icon

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
icons

bys1 committed:
merge

bys1 committed:
merge

kxv26 committed:
finished tooltips and prepared file processing for demo

kxv26 committed:
finished tooltips and prepared file processing for demo

kxv26 committed:
resources.py updated

kxv26 committed:
sorted resources

bys1 committed:
windows

bys1 committed:
small

dominictherattil committed:
Window now resizable

dominictherattil committed:
added file menu drop down and removed old method

kxv26 committed:
shortcut open/save added

kxv26 committed:
started drag and drop

kxv26 committed:
pop up before cancelling

Drovis committed:
parameter reset btn

kxv26 committed:
start on identification tabs

kxv26 committed:
start on identification tabs

Drovis committed:
remove all files button

Drovis committed:
Merge branch 'main' of https://github.com/tudor-dragan/PASTAQ-GUI

kxv26 committed:
conflict fml

kxv26 committed:
Merge branch 'main' of github.com:tudor-dragan/PASTAQ-GUI

kxv26 committed:
set starting window size

kxv26 committed:
now it actually works

kxv26 committed:
file processing with minimal hardcoding

kxv26 committed:
clean-up

kxv26 committed:
fixes

mteresapcrf committed:
Cancel button

dominictherattil committed:
fixes

bys1 committed:
Delete python

bys1 committed:
Delete sonar-properties.properties

bys1 committed:
Delete build directory

bys1 committed:
Delete .scannerwork directory

bys1 committed:
Delete .idea directory

bys1 committed:
Delete src/__pycache__ directory

bys1 committed:
Delete app.specXXXX0

bys1 committed:
Delete test1.py

bys1 committed:
Delete test2.py

tudor-dragan committed:
Added architecture and requirements documents

tudor-dragan committed:
Delete Requirements_Document.pdf

LarsSven committed:
Merge pull request #1 from tudor-dragan/submit

Submit

dominictherattil committed:
Source code for latex files

kxv26 committed:
Delete Requirements Document.zip

kxv26 committed:
Delete Architecture Document.zip

dominictherattil committed:
Added latex source code

LarsSven committed:
Merge pull request #2 from tudor-dragan/submit

Added latex source code

mteresapcrf committed:
Grey Color added

dominictherattil committed:
dark/light mode complete

dominictherattil committed:
Merge branch 'main' of https://github.com/tudor-dragan/PASTAQ-GUI

kxv26 committed:
fixed dark mode

kxv26 committed:
error handling

kxv26 committed:
fixed more dark mode stuff

kxv26 committed:
reimplemented previous attempt of drag and drop

bys1 committed:
Delete PASTAQ-GUI.spec

bys1 committed:
Delete pastaqgui.spec

bys1 committed:
Delete setup.py

bys1 committed:
Update build.yml

bys1 committed:
cleanup

bys1 committed:
Create pastaqgui.spec

bys1 committed:
Create PASTAQ-GUI.spec

bys1 committed:
test

bys1 committed:
test2

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
test

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Delete PASTAQ-GUI.spec

bys1 committed:
Delete pastaqgui.spec

bys1 committed:
Update build.yml

bys1 committed:
Update build.yml

bys1 committed:
Merge pull request #3 from tudor-dragan/main

work

bys1 committed:
Update build.yml

bys1 committed:
Delete src/__pycache__ directory

bys1 committed:
Delete resources.py

bys1 committed:
Update build.yml

dominictherattil committed:
collapsible parameters wip (currently frozen)

dominictherattil committed:
file removal

kxv26 committed:
sonarqube + resources import

kxv26 committed:
Delete resources.py

kxv26 committed:
testing gitignore

kxv26 committed:
Delete .idea directory

kxv26 committed:
changed gitignore

kxv26 committed:
Delete identification.py

kxv26 committed:
cleanup import

kxv26 committed:
edited instructions

tudor-dragan committed:
Threading issue update

Drovis committed:
drag and drop with file filter

bys1 committed:
params

bys1 committed:
pipeline

bys1 committed:
files

bys1 committed:
params

bys1 committed:
files

kxv26 committed:
clean up parameter file, more error handling and identification process logic and changed shortcut

kxv26 committed:
clean up pipeline file

kxv26 committed:
started on parameter file

tudor-dragan committed:
Documentation start

dominictherattil committed:
Pop up for reseting parmeters

kxv26 committed:
all kinds of maintenance, state management, client suggestions, refactoring

bys1 committed:
Create test.yml

bys1 committed:
Update test.yml

bys1 committed:
Create test.py

bys1 committed:
Update test.yml

kxv26 committed:
refactoring of parameter.py

kxv26 committed:
refactoring of app.py

kxv26 committed:
cleanup

tudor-dragan committed:
Documented the code for future developers - preventive maintenance

kxv26 committed:
imports adjusted to sonarqube standards and fixed bugs - corrective maintenance

Drovis committed:
select all function instead of remove all

kxv26 committed:
adjusted to sonarqube standards - corrective maintenance

ci-chriscobb committed:
fixed new project error - corrective maintenence

tudor-dragan committed:
Started testing for params file - preventive maintenance

tudor-dragan committed:
Started testing params file - preventive maintenance

kxv26 committed:
Link to guide in GUI, correcting typos, config generation, saveparams path in project, error handling - corrective and perfective maintenance

kxv26 committed:
edit file process clarified for users - perfective maintenance'

tudor-dragan committed:
Testing parameter file - preventive maintenance

tudor-dragan committed:
Merge branch 'working' of https://github.com/tudor-dragan/PASTAQ-GUI into working

kxv26 committed:
Error handling, added shortcuts - corrective and perfective maintenance

kxv26 committed:
Merge remote-tracking branch 'origin/working' into working

kxv26 committed:
removed circular import - corrective maintenance

bys1 committed:
test files

bys1 committed:
Update test.yml

bys1 committed:
Update test.yml

bys1 committed:
CI tests

bys1 committed:
CI tests

bys1 committed:
CI tests

bys1 committed:
Update test.yml

tudor-dragan committed:
Added new parameter tests - preventive maintenance

tudor-dragan committed:
Merge branch 'working' of https://github.com/tudor-dragan/PASTAQ-GUI into working

bys1 committed:
test files

bys1 committed:
files

bys1 committed:
files

tudor-dragan committed:
Added tests for the parameters.py file for the ParametersWidget class - preventive maintenance

kxv26 committed:
implemented lecturers suggestions, added more information, more tooltips, fixed bug, links to external technologies - corrective and preventive maintenance

kxv26 committed:
added logo, splash screen, more user-friendly parameters tab - corrective maintenance

kxv26 committed:
inform mac of unavailable processing - perfective maintenance

kxv26 committed:
checkboxes for parameter categories - perfective maintenance

kxv26 committed:
higher quality icons - perfective maintenance

kxv26 committed:
alternating shadow for table - perfective maintenance

bys1 committed:
Compile resources for CI tests - preventive maintenance

bys1 committed:
try to fix import - corrective maintenance

kxv26 committed:
removed circular import - corrective maintenance

kxv26 committed:
Merge branch 'working' of github.com:tudor-dragan/PASTAQ-GUI into working

bys1 committed:
file testing - preventive maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
file refactoring - preventive maintenance

tudor-dragan committed:
Testing for parameters.pt - preventive maintenance

tudor-dragan committed:
"added comments - perfective maintenance"

tudor-dragan committed:
fixed merge conflict

tudor-dragan committed:
Finished testing params - preventive maintenance

kxv26 committed:
splash screen - perfective maintenance

Drovis committed:
test 1 app.py

kxv26 committed:
testing of check_path in files - preventitive maintenance

ci-chriscobb committed:
Testing for pipeline.py - preventive maintenance

bys1 committed:
params tests

bys1 committed:
Add test dependencies - preventive maintenance

bys1 committed:
tests with pyqt5 - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
pyqt5 tests - preventive maintenance

bys1 committed:
imports - preventive maintenance

bys1 committed:
pytest - preventive maintenance

bys1 committed:
mock - preventive maintenance

bys1 committed:
testing - preventive maintenance

kxv26 committed:
files.py testing done -preventive maintenance

bys1 committed:
testing - preventive maintenance

bys1 committed:
Merge branch 'working' of github.com:tudor-dragan/PASTAQ-GUI into working

bys1 committed:
file tests - preventive maintenance

bys1 committed:
TEST

bys1 committed:
cleanup - preventive maintenance

tudor-dragan committed:
Corrected temporary directory for tests to be able to run on any machine - corrective maintenance

kxv26 committed:
coverage for sonarqube and cleanup of test_files - preventive maintenance

kxv26 committed:
scanner added to gitignore -preventive maintenance

kxv26 committed:
Delete .scannerwork directory

kxv26 committed:
Delete main.py

kxv26 committed:
fixed imports -  preventive maintenance

kxv26 committed:
pipeline testing - preventive maintenance

bys1 committed:
test pipelines - preventive maintenance

tudor-dragan committed:
fixed code smells in test_params.py - corrective maintenance

kxv26 committed:
made all testing work, removede code smells - corrective maintenance

Drovis committed:
app test v2

kxv26 committed:
fixed layout to scale description and parameters - corrective maintenance

bys1 committed:
run button - perfective maintenance

bys1 committed:
fix index error - corrective maintenance

kxv26 committed:
exception to error changed in files.py reflected in testing as well - corrective maintenance

tudor-dragan committed:
Added identifiers for tests - perfective maintenance

bys1 committed:
Delete .DS_Store

bys1 committed:
testing - preventive maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
import - corrective maintenance

bys1 committed:
remove menu button, fix shortcuts - perfective maintenance

bys1 committed:
cleanup files - preventive maintenance

bys1 committed:
tests CI - preventive maintenance

tudor-dragan committed:
fixed params test import

kxv26 committed:
fixed dark mode

dominictherattil committed:
fixed light mode

kxv26 committed:
fixed dark mode indefinitely, added drop indication for user, added more information to paths tab, made titles bold to stand out more, changed pramaters tab to non-scrollable since minimum size allows for not scrollable, changed button naming to something more informative - prefective and corrective maintenance

kxv26 committed:
changed logging of running - corrective maintenance

bys1 committed:
path checks - perfective maintenance

tudor-dragan committed:
Changed placeholder error - corrective maintenance

Drovis committed:
app testing v2

Drovis committed:
Merge branch 'working' of https://github.com/tudor-dragan/PASTAQ-GUI into working

bys1 committed:
path checks - perfective maintenance

Drovis committed:
test_app.y test labeling

kxv26 committed:
fixed drag and drop, added more feedback for edit dialog, updated quality of png for smaller size -corrective and perfective maintenance

bys1 committed:
cleanup and icons - perfective maintenance

kxv26 committed:
more descriptive paths description - perfective maintenance

bys1 committed:
popup - corrective maintenance

bys1 committed:
Merge branch 'working' of github.com:tudor-dragan/PASTAQ-GUI into working

kxv26 committed:
more information for the user - corrective maintenance

kxv26 committed:
more specific information and captions - corrective maintenance

kxv26 committed:
50% of testing of config - perfective maintenance

kxv26 committed:
finished app testing - perfective maintenance

tudor-dragan committed:
Fixed popup appearing while testing and few tests not running in params - corrective maintenance

tudor-dragan committed:
Merge branch 'working' of https://github.com/tudor-dragan/PASTAQ-GUI into working

kxv26 committed:
fixed testing - corrective maintenance

kxv26 committed:
finished testing

bys1 committed:
test - corrective maintenance

bys1 committed:
Remove Linux executable - perfective maintenance

bys1 committed:
fix test - corrective maintenance

bys1 committed:
cleanup - preventive maintenance

bys1 committed:
cleanup - preventive maintenace

ci-chriscobb committed:
corrective maintenance - changing unit test identifiers

ci-chriscobb committed:
corrective maintenance - reverting previous commit

kxv26 committed:
deliverable for client

kniren commented:
Hello,

This is a big PR, which I will accept with a couple of changes, since we have been talking about it for a number of months. For next time, however, I recommend splitting the PR into different cherry-picked ones (e.g. one PR for adding CI support, another for refactoring in multiple files, adding some sort of functionality, etc.).

Two changes are requested before I merge this:
1) Make sure you are not using copyrighted material (for example the images for the question mark or the logo) and replace if needed.
2) Update the readme with instructions about how to build and run this application.

Best,
Alex

tudor-dragan committed:
Updated README with resource generation

tudor-dragan committed:
Changed tooltip icon to non-copyrighted one

tudor-dragan commented:
I have updated the readme with the command to install `pyrcc5` and also to generate the resources with it. I have replaced the icon with a non-copyrighted one we created.
